homeassistant/components/ridwell/calendar.py

```diff
@@ -4,31 +4,58 @@
 
 import datetime
 
-from aioridwell.model import RidwellAccount, RidwellPickupEvent
+from aioridwell.model import PickupCategory, RidwellAccount, RidwellPickupEvent
 
 from homeassistant.components.calendar import CalendarEntity, CalendarEvent
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import DOMAIN
+from .const import (
+    CALENDAR_TITLE_NONE,
+    CALENDAR_TITLE_ROTATING,
+    CALENDAR_TITLE_STATUS,
+    CONF_CALENDAR_TITLE,
+    DOMAIN,
+)
 from .coordinator import RidwellDataUpdateCoordinator
 from .entity import RidwellEntity
 
 
 @callback
 def async_get_calendar_event_from_pickup_event(
-    pickup_event: RidwellPickupEvent,
+    pickup_event: RidwellPickupEvent, config_entry: ConfigEntry
 ) -> CalendarEvent:
     """Get a HASS CalendarEvent from an aioridwell PickupEvent."""
-    pickup_type_string = ", ".join(
-        [
-            f"{pickup.name} (quantity: {pickup.quantity})"
-            for pickup in pickup_event.pickups
-        ]
-    )
+    pickup_items = []
```

Comment:
You can streamline the creation of this list with a comprehension:

```python
pickup_items = [f"{pickup.name} (quantity: {pickup.quantity})" for pickup in pickup_event.pickups]
```

Reply:
This means you'll need to calculate `rotating_category` differently, but it, too, can be streamlined:

```python
rotating_category = next((pickup.name for pickup in pickup_event.pickups if pickup.category == PickupCategory.ROTATING), "")
```